S4015: Migrate to LAYC level 2 (#2134)
SonarSource/rspec: rules/S4015/csharp/rule.adoc

== Why is this an issue?

> Changing an inherited method that is not an https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/override[override] to https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private[private] will not prevent access to the base class implementation. Furthermore, having a derived class shadow the name of a base method can lead to confusion.

Reviewer: I am not sure of the relevance of this:
> ...will not prevent access to the base class implementation.
I think it can be removed, as it might not be why they changed the access modifier.

My suggestions:
```suggestion
Decreasing the https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers[accessibility level] of an inherited method that is not https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/override[overridable] to https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private[private] will shadow the name of the base method and can lead to confusion.
```
